Public/Src/Tools/VBCSCompilerLogger/VBCSCompilerLogger.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using System.Linq;
+using BuildXL.Native.IO;
+using BuildXL.Processes;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.CodeAnalysis;
+
+namespace VBCSCompilerLogger
+{
+    /// <summary>
+    /// This logger catches csc and vbc MSBuild tasks and uses the command line argument passed to the compiler to mimic the file accesses that the compiler 
+    /// would have produced.
+    /// </summary>
+    /// <remarks>
+    /// This MSBuild logger is used by the MSBuild frontend of BuildXL when scheduling pips: when process breakaway is supported, VBSCompiler.exe is allowed to escape the sandbox
+    /// and outlive the originating pip. This logger is used as a way to compensate for the missing accesses and the MSBuild frontend attaches it to every MSBuild invocation.
+    /// </remarks>
+    public class CompilerFileAccessLogger : Logger
+    {
+        private const string CscTaskName = "Csc";
+        private const string VbcTaskName = "Vbc";
+        private const string CscToolName = "csc.exe";
+        private const string VbcToolName = "vbc.exe";
+        private AugmentedManifestReporter m_augmentedReported;

```

Comment:
typo: name should end in Reporter right?

Reply:
yes

---
In reply to: [346068547](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346068547) [](ancestors = 346068547)